sdk/monitor/Azure.Monitor.Query/src/MetricsBatchQueryClient.cs

```diff
@@ -0,0 +1,214 @@
+        /// <returns></returns>
+        public virtual Response<MetricResultsResponse> Batch(List<string> resourceIds, List<string> metricNames, string metricNamespace, MetricsQueryOptions options = null, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsBatchQueryClient)}.{nameof(Batch)}");
+            scope.Start();
+
+            try
+            {
+                if (resourceIds.Count == 0)
+                {
+                    throw new ArgumentException("Resource IDs can not be empty");
+                }
+
+                var subscriptionId = GetSubscriptionId(resourceIds[0]);
+
+                string filter = null;
+                TimeSpan? granularity = null;
+                string aggregations = null;
+                string startTime = null;
+                int? top = null;
+                string orderBy = null;
+                string endTime = null;
+                ResourceIdList resourceIdList = new ResourceIdList(resourceIds);
+
+                if (options != null)
+                {
+                    startTime = options.TimeRange.Value.Start.ToString();
+                    endTime = options.TimeRange.Value.End.ToString();
+
+                    top = options.Size;
+                    orderBy = options.OrderBy;
+                    filter = options.Filter;
+                    granularity = options.Granularity;
+                }
+
+                return _metricBatchClient.Batch(
+                    subscriptionId,
+                    metricNamespace,
+                    metricNames,
+                    resourceIdList,
+                    startTime,
+                    endTime,
+                    granularity,
+                    aggregations,
+                    top,
+                    orderBy,
+                    filter,
+                    cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private string GetSubscriptionId(string resourceId)
+        {
+            int startIndex = resourceId.IndexOf("subscriptions/") + 14;
+            return resourceId.Substring(startIndex, resourceId.IndexOf("/", startIndex) - startIndex);
+        }
+
+        /// <summary>
+        /// todo
+        /// </summary>
+        /// <param name="resourceIds"></param>
+        /// <param name="metricNames"></param>
+        /// <param name="metricNamespace"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException"></exception>
+        public virtual async Task<Response<MetricResultsResponse>> BatchAsync(List<string> resourceIds, List<string> metricNames, string metricNamespace, MetricsQueryOptions options = null, CancellationToken cancellationToken = default)
```

Comment:
It would be good to have a private method that can be used for both sync/async that has an `async` parameter to avoid duplicating code.